src/NServiceBus.Core/InitializableEndpoint.cs

```diff
@@ -182,5 +183,6 @@ string GetInstallationUserName()
         PipelineConfiguration pipelineConfiguration;
         PipelineSettings pipelineSettings;
         SettingsHolder settings;
+        CriticalError criticalError;
```

Comment:
what about `RegisterCriticalErrorHandler` returning the `criticalErrorHandler` instead of using a field? Looks like that field is only used within `Initialize` at first glance.